department-of-veterans-affairs/vets-website
# CST | Fix issue with incorrect page count when paginating

## Original Issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/44111

## URL of change
[/track-claims/your-claims](http://localhost:3001/track-claims/your-claims)

## Background
Please describe the context critical to your pull request. Ensure that you use enough context that someone who is not familiar with your application can read this and understand what is being added and why.

## Steps to test
NOTE: To see the issue you should replace your `/src/applications/claims-status/tests/e2e/fixtures/mock/mock-api/claims/claims-list.json` with: [claims-list.json.zip](https://github.com/department-of-veterans-affairs/vets-website/files/9095641/claims-list.json.zip)

* Login with a staging user and navigate to: [/track-claims/your-claims](http://localhost:3001/track-claims/your-claims)
* There should be a section labeled 'Your claims and appeals'
* You should see a list of cards and the text `Showing 1-10 of 21 events`
* If you scroll to the bottom of the list you should see a `va-pagination` component with a `1` and a `2`, but no `3`

## Code changes
### Core
* Moved `page` and `pages` management out of Redux and into the component
* Removed action creators that were no longer needed because of the above

### Miscellaneous
* `VaPagination` is only added to DOM when needed
* Removed `p` tags that were unneeded
* Replaced `div` with `<></>`
* Updated unit tests

## How was your code tested
Tests still pass

## Acceptance criteria
- [x] Pagination displays correct number of pages
